pandas/io/excel.py

```diff
@@ -202,6 +202,16 @@ def read_excel(io, sheet_name=0, header=0, skiprows=None, skip_footer=0,
                dtype=None, true_values=None, false_values=None, engine=None,
                squeeze=False, **kwds):
 
+    # Can't use _deprecate_kwarg since sheetname=None has a special meaning
+    if sheet_name is 0 and 'sheetname' in kwds:
```

Comment:
I think we want that here. `sheet_name` could be a str, int, or None and we don't want to raise a TypeError. Just checking whether they changed the default.